gensim/models/keyedvectors.py

```diff
@@ -158,264 +163,425 @@
 
-    """
-    def __init__(self, **kwargs):
-        self.count = 0
-        self.__dict__.update(kwargs)
 
-    def __lt__(self, other):  # used for sorting in a priority queue
-        return self.count < other.count
-
-    def __str__(self):
-        vals = ['%s:%r' % (key, self.__dict__[key]) for key in sorted(self.__dict__) if not key.startswith('_')]
-        return "%s(%s)" % (self.__class__.__name__, ', '.join(vals))
+class KeyedVectors(utils.SaveLoad):
+    def __init__(self, vector_size, mapfile_path=None):
+        """Mapping between keys (such as words)  and vectors for :class:`~gensim.models.Word2Vec`
+        and related models.
 
+        Used to perform operations on the vectors such as vector lookup, distance, similarity etc.
 
-class BaseKeyedVectors(utils.SaveLoad):
-    """Abstract base class / interface for various types of word vectors."""
-    def __init__(self, vector_size):
-        self.vectors = zeros((0, vector_size), dtype=REAL)
-        self.vocab = {}
+        """
         self.vector_size = vector_size
-        self.index2entity = []
+        self.vectors = zeros((0, vector_size), dtype=REAL)  # fka (formerly known as) syn0
+        self.norms = None
 
-    def save(self, fname_or_handle, **kwargs):
-        super(BaseKeyedVectors, self).save(fname_or_handle, **kwargs)
+        self.index_to_key = []  # fka index2entity or index2word
+        self.key_to_index = {}
 
-    @classmethod
-    def load(cls, fname_or_handle, **kwargs):
-        return super(BaseKeyedVectors, cls).load(fname_or_handle, **kwargs)
+        self.expandos = {}  # dynamically-expandable per-vector named, numpy-typed attributes
+
+        self.mapfile_path = mapfile_path
+
+    def _load_specials(self, *args, **kwargs):
+        """Handle special requirements of `.load()` protocol, usually up-converting older versions."""
+        super(KeyedVectors, self)._load_specials(*args, **kwargs)
+        if hasattr(self, 'doctags'):
+            self._upconvert_old_d2vkv()
+        # fixup rename/consolidation into index_to_key of older index2word, index2entity
+        if not hasattr(self, 'index_to_key'):
+            self.index_to_key = self.__dict__.pop('index2word', self.__dict__.pop('index2word', None))
+        # fixup rename into vectors of older syn0
+        if not hasattr(self, 'vectors'):
+            self.vectors = self.__dict__.pop('syn0', None)
+            self.vector_size = self.vectors.shape[1]
+        # ensure at least a 'None' in 'norms' to force recalc
+        if not hasattr(self, 'norms'):
+            self.norms = None
+        # ensure at least an empty 'expandos'
+        if not hasattr(self, 'expandos'):
+            self.expandos = {}
+        # fixup rename of vocab into map
+        if 'key_to_index' not in self.__dict__:
+            self._upconvert_old_vocab()
+
+    def _upconvert_old_vocab(self):
+        """Convert a loaded, prior-version instance that had a 'vocab' dict of data objects"""
```

Comment:
```suggestion
        """Convert a loaded, prior-version instance that had a 'vocab' dict of data objects."""
```

What's "prior-version"?

Reply:
All pre-this-PR KV types had some sort of `.vocab` property. I'll reword the comment to mention "pre-4.0.0".